File: src/main/java/aurocosh/divinefavor/common/util/CustomTeleporter.java
package aurocosh.divinefavor.common.util;

import net.minecraft.entity.Entity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.util.math.Vec3d;
import net.minecraft.world.Teleporter;
import net.minecraft.world.WorldServer;

public class CustomTeleporter extends Teleporter {
    private final WorldServer worldServerInstance;
    private Vec3d destination;

    public CustomTeleporter(WorldServer world, Vec3d destination) {
        super(world);
        this.worldServerInstance = world;
        this.destination = destination;
    }

    @Override
    public void placeInPortal(Entity pEntity, float rotationYaw) {
        this.worldServerInstance.getBlockState(new BlockPos(destination));
        pEntity.setPosition(destination.x, destination.y, destination.z);
        pEntity.motionX = 0.0f;
        pEntity.motionY = 0.0f;
        pEntity.motionZ = 0.0f;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/util/CustomTeleporter.kt
package aurocosh.divinefavor.common.util

import net.minecraft.entity.Entity
import net.minecraft.util.math.BlockPos
import net.minecraft.util.math.Vec3d
import net.minecraft.world.Teleporter
import net.minecraft.world.WorldServer

class CustomTeleporter(private val worldServer: WorldServer, private val destination: Vec3d) : Teleporter(worldServer) {
    override fun placeInPortal(entity: Entity, rotationYaw: Float) {
        this.worldServer.getBlockState(BlockPos(destination))
        entity.setPosition(destination.x, destination.y, destination.z)
        entity.motionX = 0.0
        entity.motionY = 0.0
        entity.motionZ = 0.0
    }
}
